xds/src/main/java/io/grpc/xds/orca/OrcaMetricReportingServerInterceptor.java

```diff
@@ -46,27 +47,40 @@
 @ExperimentalApi("https://github.com/grpc/grpc-java/issues/9127")
 public final class OrcaMetricReportingServerInterceptor implements ServerInterceptor {
 
-  private static final OrcaMetricReportingServerInterceptor INSTANCE =
-      new OrcaMetricReportingServerInterceptor();
+  private static volatile OrcaMetricReportingServerInterceptor instance;
 
   @VisibleForTesting
   static final Metadata.Key<OrcaLoadReport> ORCA_ENDPOINT_LOAD_METRICS_KEY =
       Metadata.Key.of(
           "endpoint-load-metrics-bin",
           ProtoUtils.metadataMarshaller(OrcaLoadReport.getDefaultInstance()));
 
-  private volatile MetricRecorder metricRecorder;
+  private final MetricRecorder metricRecorder;
 
   @VisibleForTesting
-  OrcaMetricReportingServerInterceptor() {
+  OrcaMetricReportingServerInterceptor(MetricRecorder metricRecorder) {
+    this.metricRecorder = metricRecorder;
+    OrcaMetricReportingServerInterceptor.instance = this;
   }
 
-  public static OrcaMetricReportingServerInterceptor getInstance() {
-    return INSTANCE;
+  /**
+   * Returns the server interceptor if created, otherwise {@code null}.
+   */
+  @Nullable
+  public static OrcaMetricReportingServerInterceptor getInstanceIfCreated() {
+    return instance;
   }
 
-  public void setMetricRecorder(MetricRecorder mr) {
-    metricRecorder = mr;
+  public static OrcaMetricReportingServerInterceptor getOrCreateInstance(
+      MetricRecorder metricRecorder) {
+    if (instance == null) {
```

Comment:
No, you should be able to create more than one of these interceptors. Create a new instance every time in this method. You can name it `create(MetricRecorder)` if you want.

Reply:
Done. Renamed to `create(MetricRecorder)` and create a new instance everytime.